client/src/app/deck/deck-select.tsx

```diff
@@ -61,12 +61,14 @@ function SelectDeck({
 
 	// STATE
 	const [savedDecks, setSavedDecks] = useState<Array<string>>(getSavedDecks)
-	const sortedDecks = savedDecks
-		.map((d: any) => {
+	function parseDecks(decks: string[]): Array<PlayerDeckT> {
```

Comment:
This seems extremely overcomplicated to me. We're saving the decks as strings and continuously reparsing them. What should really be happening is we should have a tracked state for a PlayerDeckT array that gets updated using the useState stuff, and every time a change is made we should stringify and save decks to local storage. We should not need to parse from strings other than when we first load.

This was probably always set up wrong tbh. This whole thing could be simplified a lot, but right now it works so therefore I think we should just merge it, and then redo the whole file at a later date.